### PR TITLE
Add OCR recognition in Flutter, process-image edge function and link between the 2

### LINKED ISSUE #140: feat: add OCR recognition & AI processing for images

### PR DESCRIPTION
**Pull request by pmdlt**
**Description:**
This PR adds OCR (Optical Character Recognition) and AI processing for images to our Flutter application. With this enhancement, users can now extract text from images, particularly useful for scenarios like scanning newspapers or other printed materials. The extracted text is then passed through an AI model to provide additional context or information based on the extracted text.

**Changes:**
- Added `NewsRecognitionViewModel` to handle text recognition and AI processing.
- Integrated Google ML Kit Text Recognition for extracting text from images.
- Implemented Supabase edge function `process-image` to process the extracted text using OpenAI's GPT-3.5 model.
- Updated `pubspec.yaml` to include dependencies for text recognition and camera.
- Added TypeScript code for the Supabase edge function `process-image`.
- Linked the issue #140 that will be closed by this pull request.

**Testing:**
- Tested the OCR recognition and AI processing functionality with various types of images containing text.
- Verified that the extracted text is correctly processed by the AI model and relevant information is provided.

**Related Issue:**
Closes #140

### CODE REVIEW

File: supabase/functions/process-image/index.ts
Comment (ClementJosso) [inline supabase/functions/process-image/index.ts:1]: I am not familiar with typescript so I trust you on the functionality of this function, however I would like to ask what it outputs. Does it write some sort of long string of text corresponding to the response of the LLM directly into the DB ?

**Review by ClementJosso — COMMENTED** [review]
The view model code looks good and I trust that the edge function works as well, I am not approving yet just to make sure that the bug I have with the mlkit import (mentioned in a previous comment) is only on my machine and not a systematic problem in the team.

File: lib/viewmodels/news_recognition.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:google_mlkit_text_recognition/google_mlkit_text_recognition.dart';
```
[the remark this answers is not in the document]
Comment (pmdlt) [reply lib/viewmodels/news_recognition.dart:4]: You need to `flutter pub get` maybe ?

[the remark this answers is not in the document]
Comment (ClementJosso) [reply lib/viewmodels/news_recognition.dart:4]: You were right it was as simple as that

File: supabase/functions/process-image/index.ts
Comment (pmdlt) [reply supabase/functions/process-image/index.ts:1]: Juste a simple text :) string yes

File: supabase/functions/process-image/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const userId = user.data.user.id;
+
+  const openai = new OpenAI();
+  const completion = await openai.chat.completions.create({
+    "model": "gpt-3.5-turbo",
+    "messages": [
+      {
+        "role": "system",
+        "content":
+          "The user provides you an OCR text (not necessarily perfectly formatted) of a photo from a newspaper, or some other news item. The news is not recent, so you should be able to find it. If this doesn't mean anything to you, simply return a message saying you don't know the facts. If you do know the facts, then you give as much context as you can on them, relate the historical facts, the people involved, and so on.",
```
Comment (Pacsort17) [inline supabase/functions/process-image/index.ts:52]: "The news is not recent" is not usefull I think

**Review by Pacsort17 — DISMISSED** [review]
I couldn't test the AI on my side, or at least I didn't know how.

However the script seems ok, some parts can be removed.

File: supabase/functions/process-image/index.ts (hunk 2 above)
Comment (pmdlt) [reply supabase/functions/process-image/index.ts:52]: When I made several attempts, I regularly came across the message "I'm not able, I'm stopping at 2020 etc." and with this message, I got much better answers as if the AI took into account the fact that it was in the past.

Comment (pmdlt) [reply supabase/functions/process-image/index.ts:52]: So I think it's worth keeping, don't you?

**Review by ClementJosso — APPROVED** [review]
Thank you for your response to my comments, I have nothing more to say :)

**Pull request merged**